openmls/src/framing/sender.rs

```diff
@@ -61,20 +66,71 @@ impl TryFrom<u8> for SenderType {
     }
 }
 
-#[derive(
-    PartialEq, Clone, Copy, Debug, Serialize, Deserialize, TlsDeserialize, TlsSerialize, TlsSize,
-)]
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub enum SenderValue {
+    Member(KeyPackageRef),
+    Preconfigured(TlsByteVecU8),
+    NewMember,
+}
+
+#[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub struct Sender {
     pub(crate) sender_type: SenderType,
-    // TODO: #541 replace sender with [`KeyPackageRef`] (and preconfigured/new)
-    pub(crate) sender: LeafIndex,
+    pub(crate) sender: SenderValue,
 }
 // Public functions
 impl Sender {
+    /// Build a [`Sender`] from [`MlsSenderData`].
+    pub(crate) fn from_sender_data(sender_data: MlsSenderData) -> Self {
```

Comment:
Would it make sense to implement `From` for `MlsSenderData` and `&KeyPackageRef` instead of these explicit functions?

Reply:
We could. I made them explicit because there has been confusion around conversation a lot recently. Having to type it makes it clearer what happens. I don't have strong feelins on this but wanted to be a little more explicit to see if it helps.